htdocs/compta/facture/card.php

```diff
@@ -2668,7 +2668,11 @@
 
 		$line = new FactureLigne($db);
 		$line->fetch(GETPOSTINT('lineid'));
-		$percent = $line->get_prev_progress($object->id);
+		if (getDolGlobalInt('INVOICE_USE_SITUATION') == 2) {
+			$percent = 0; // New situation percent must be 0 (No cumulative)
+		} else {
```

Comment:
Percent is calculated after ($addprogress on line 2764)
The problem is on the updateline() method.

```
// Line 4227
if (!isset($situation_percent) || $situation_percent > 100 || (string) $situation_percent == '' || $situation_percent == null) {
	$situation_percent = 100;
}
```

The fix must be the same as the addline method
```
if (!isset($situation_percent) || $situation_percent > 100 || (string) $situation_percent == '' || $situation_percent == null) {
	// INVOICE_USE_SITUATION = 2 - Lines situation percent on new lines must be 0 (No cumulative)
	if ($this->isSituationInvoice() && getDolGlobalInt('INVOICE_USE_SITUATION') == 2) {
		$situation_percent = 0;
	} else {
		$situation_percent = 100;
	}
}
```

Reply:
but as i merge your fix i don't understand why your code is not "visible here" ...

Reply:
ok i understand, you put a comment on card.php and the fix is on the other file .class.php ...